Dockerfile

```diff
@@ -0,0 +1,98 @@
+ARG RUBY_VERSION
+ARG ALPINE_VERSION=3.13
+
+###############################################################################
+# Stage 1 - Base Image
+###############################################################################
+FROM ruby:$RUBY_VERSION-alpine$ALPINE_VERSION AS base
+
+# Fix vulnerability of version  of the apk-tools package
+RUN apk add --update apk-tools=2.12.5-r0
+
+# Installed dependencies used in the build and production image
+RUN apk add --update \
+  postgresql-client \
+  tzdata \
+  nodejs \
+  yarn
+
+# Install specific version of bundler
+RUN gem install bundler -v 2.2.17
+
+ARG USER=rails
+ARG UID=1000
+ARG GID=1000
+
+RUN addgroup -g $GID $USER && adduser -D -u $UID -G $USER $USER
+
+ARG USER=rails
+USER $USER
+###############################################################################
+# Stage 2 - Test Image
+###############################################################################
+FROM base AS test
+
+USER root
+
+RUN apk add --update \
+  postgresql-dev \
+  sqlite \
+  sqlite-dev \
+  build-base
+
+WORKDIR /usr/src/myapp
+
+COPY . .
+RUN bundle _2.2.17_ config set with 'test'
+RUN bundle _2.2.17_ install --jobs=4 --retry=3
+RUN RAILS_ENV=test RAKE_ENV=test bundle _2.2.17_ exec rake -T test
```

Comment:
This line is incorrect, it does not run tests. It should be replaced with `RUN RAILS_ENV=test RAKE_ENV=test bundle _2.2.17_ exec rake test`

Reply:
You are correct. I've updated the `test` task so that it actually runs the tests.

Also, I've updated the `Dockerfile` to use a cache for the `vendor` assets and to simplify asset pre-compilation.